934fe8c/README.md

```diff
@@ -283,6 +283,153 @@ The trusted-setup coordinator will give each participant an `upload_url` which c
 $ rsync -vP --append -e 'ssh -i <path to SSH private-key>' <files> <upload url>
 ```
 
+## Phase 2 Validation
+
+**Note**: The verification process is very computationally intensive. For details see in the [Hardware Requirements](#hardware-requirements) section the "Contribution Verification" requirements.
+
+1. Build `phase2`. Please follow the instruction from the [Install Dependencies and Build `phase2`](#03-install-dependencies-and-build-phase2) section and stop right before the "Build `phase2`" step. Instead do the following:
+
+```console
+$ git clone https://github.com/filecoin-project/filecoin-phase2.git
+$ cd filecoin-phase2
+$ git checkout 934fe8c6d2df2589644302579838976d070c48a7
+$ cargo build --release --bins
+$ mv target/release/filecoin-phase2 phase2
+```
+
+2. Download verification scripts:
+
+```console
+TODO vmx 2022-01-20
+```
+
+3. Verify Phase2 contributions:
+
+In order to run the verification, you will need to download the relevant Phase 1 file for the circuit and run the verification script for that specific circuit. The verification script will verify all Phase 2 contributions made to the Groth16 parameters for a circuit of that specific size and check that the verified Groth16 parameters are being used by the Filecoin network. The commands and associated Phase 1 file for each circuit are listed below.
+
+All files that need to be downloaded for the verification process are stored on IPFS with root hash `QmbcqEiRyqe9sq9zxw7b6obuGw1vpVe7SzgZ6U5CoZYRpZ` and are also available at https://trusted-setup-snapdeals.filecoin.io/.
+
+Running `verify_all.sh` as illustrated here is very time-consuming. These commands are provided for reference, to clarify what must be verified. In practice, we recommend that verifiers follow the instructions in the next section, **Verify all contributions individually**.
+
+**Empty sector upgrade**
+
+```console
+$ ./verify_all.sh update 32
+$ ./verify_all.sh poseidon 64
```

Comment:
Would it be possible to use `updatep` instead of `poseidon` as the argument name? Just for consistency with the rest of the trusted-setup.

Reply:
That was intentional. I wanted to prevent typos. It's too easy to get `update` and `updatep` wrong. I wanted to make it for folks that might want to verify it easier. Though if you feel strongly about it, I can also change it.

Reply:
Sounds good to me